### PR TITLE
Sponsorblock integration

### LINKED ISSUE #1: SponsorBlock support
[SponsorBlock](https://sponsor.ajay.app/) is an open-source crowdsourced browser extension and open API for skipping sponsor segments in YouTube videos.

They [seemingly support youtube.com/.tv](https://github.com/ajayyy/SponsorBlock/pull/701) so that should be pretty possible to just pull in their contentscript, after beating their ts(x) with enough webpack/babel magic.

### LINKED ISSUE #11: Slow DIAL connection
Needs further research, but seems like after changes in e6d7c03dccb585426ef6f1a18ae3b4882df690d2 (needs verification) DIAL connection takes approximately 30-40 seconds after app launch to establish.

At first, connection to http://localhost:18181 was getting rejected due to missing CORS headers, but after adding `{"vendorExtension": {"allowCrossDomain": true}}` as per original manifest, the issue still persists... YouTube TV attempts to send a GET request, which is unhandled and stays pending for a while...

### PR DESCRIPTION
**Pull request by Informatic**
This is a very early implementation of sponsorblock integration.
* webpack+babel-based build infrastructure for userscript and index - no longer need to worry about using modern syntax
* userscript has been split into mulitple files with `userScript.js` just initializing necessary shims and pulling in the submodules.
* Segments are marked as colored bars on progress slider (this is slightly sketchy - based on MutationObserver - will likely not work on webOS 1.x/2.x - but not sure if these are even supported by YouTube TV...)
* No skip category configuration UI
* Small skip notification in bottom right corner is shown.
* Settings UI (checkboxes for adblocking and sponsorblock features, both enabled by default) is accessible after clicking [GREEN] (notification about this extra feature is shown for a second on startup)
* All categories are skipped by default
* No unskipping (though manual seek backwards should work more-or-less fine)

Closes #1
Closes #11 

c1
c2
c3

### CODE REVIEW

**Comment by FriedChickenButt** [comment]
Thank you so much for the PR, will test today once before merging.

**Comment by Informatic** [comment]
I would honestly rather have this pending for a while before merging, until we thoroughly test it out and add some sort of configuration UI.

**Comment by Informatic** [comment]
Just how I expected, a bunch of bugs already ironed out.

**Comment by Informatic** [comment]
This now has an initial configuration UI opened by pressing green button on a remote. (unused by YouTube TV by default)

D-Pad navigation is carried out by lightly patched version of https://github.com/WICG/spatial-navigation polyfill. Focus is taken over by configuration window after opening and all key events are stopped from propagaint into main UI. Back button (or pressing green button again) should close the window.

Configuration is stored as JSON in `localStorage` `ytaf-configuration` key, managed in `src/config.js`. This definitely needs some cleanups, wouldn't mind if anyone wanted to take it up for some further development :)

Tested on webOS 3.8/Chromium 38 only.

Also - it seems like our fetch/XHR hooks are ran properly, but don't *actually* block any requests in practice, at least on my tv? Maybe these rules need some refinement, or shall we just get rid of these in the end?

File: src/ui.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+    evt.preventDefault();
+    evt.stopPropagation();
+  }
+}, true);
+
+uiContainer.innerHTML = `
+<h1>webOS YouTube Extended</h1>
+<label for="__adblock"><input type="checkbox" id="__adblock" /> Enable AdBlocking</label>
+<label for="__sponsorblock"><input type="checkbox" id="__sponsorblock" /> Enable SponsorBlock</label>
```
Comment (ajayyy) [inline src/ui.js:37]:
```suggestion
<label for="__sponsorblock"><input type="checkbox" id="__sponsorblock" /> Enable SponsorBlock (https://sponsor.ajay.app)</label>
```

**Comment by ajayyy** [comment]
Nice scheduling implementation!

**Comment by Informatic** [comment]
@FriedChickenButt I think this is ready for review/testing.

**Comment by FriedChickenButt** [comment]
> This now has an initial configuration UI opened by pressing green button on a remote. (unused by YouTube TV by default) Two configuration flags for AdBlocking and SponsorBlock has been implemented and integrated into userscript logic.
> 
> D-Pad navigation is carried out by lightly patched version of https://github.com/WICG/spatial-navigation polyfill. Focus is taken over by configuration window after opening and all key events are stopped from propagaint into main UI. Back button (or pressing green button again) should close the window.
> 
> Configuration is stored as JSON in `localStorage` `ytaf-configuration` key, managed in `src/config.js`. This definitely needs some cleanups, wouldn't mind if anyone wanted to take it up for some further development :)
> 
> Tested on webOS 3.8/Chromium 38 only.
> 
> Also - it seems like our fetch/XHR hooks are ran properly, but don't _actually_ block any requests in practice, at least on my tv? Maybe these rules need some refinement to block some analytics, or shall we just get rid of these in the end?

These fetch/XHR hooks block some of the ad hosting sites. Surely, these need more refinement to block analytics.

**Comment by FriedChickenButt** [comment]
Thank you @Informatic for the contribution, tested the following:
* New App icon - Love the traffic cone/rotated youtube icon
* SponsorBlock - blocks sponsored sections and self promotion sections
* Green button to open the settings and modify the options (Cool overlay BTW)
* Yellow button for search
* HDR remains unaffected - works fine as always
* Back button now works everywhere in the app
* D-pad navigation now better and more predictable

Tested on webOS 05.00.15.

Thank you @ajayyy for reviewing the SponsorBlock code.

Merging the PR.

**Pull request merged**